### DISCUSSION TITLE
High battery drain

### DISCUSSION THREAD

**Discussion by anatskiy** [ghostty-org/ghostty#3883] [Feature Requests, Ideas] [answered]
I noticed Ghostty consumes a lot of power and macOS even tells you about it. Can this be a real issue?

<img width="346" alt="1" src="https://github.com/user-attachments/assets/f82143e7-16a4-4135-a9fa-b86f337f6efd" />

<img width="640" alt="2" src="https://github.com/user-attachments/assets/fbd7da7a-ca6e-4ee4-8096-9420aa69038d" />

I was tinkering with my tmux config this morning and noticed my MacBook's battery drained considerably during this time (by around 30%). The screen brightness was at 30%. 

<img width="468" alt="3" src="https://github.com/user-attachments/assets/518c0a8d-fea5-4b66-8d5f-a97a9098a24e" />

Currently, the Ghostty window is open in a separate virtual desktop and has a single tab with tmux. The OS still shows that Ghostty uses significant energy.

P.S. I've got MacBook Pro M1 with macOS 15.2.

**Comment by skovtunenko** [comment]
I noticed this too. 
On MacBook Pro M3 with macOS 15.2 the default IDLE state of Ghostty is:

<img width="993" alt="SCR-20241229-nqao" src="https://github.com/user-attachments/assets/f71c844f-3ac1-4d36-aac8-1e7ea933caaf" />

<img width="1365" alt="image" src="https://github.com/user-attachments/assets/1906fc03-70a4-45a6-8869-3c5fa41cb504" />

**Comment by mitchellh** [comment]
I haven't seen this myself. The CPU usage is small and when unfocused goes to zero. We also respect a lot of power saving features of macOS. If there are any tips on what triggers this I'd like to look into it.

**Comment by mitchellh** [comment]
I haven't seen this myself. The CPU usage is small and when unfocused goes to zero. We also respect a lot of power saving features of macOS. If there are any tips on what triggers this I'd like to look into it.

**Comment by w3dd1e** [comment]
Just want to add here that it's draining my batter even when I have completely closed and quit Ghostty.  I realized the energy use was high last night and shut it down at 1100, but now it's over 2000. Kept an eye on it after taking that screenshot.  it jumped up to 2021 after about 10 minutes _while fully closed_.  

<img width="1010" alt="Screenshot 2024-12-29 at 1 37 58 PM" src="https://github.com/user-attachments/assets/4cb0b49b-648e-4c85-9890-f4ffcf6d1e1a" />

**Comment by mpriscella** [comment, marked as answer]
I experienced the same issue (MacBook Pro M3, MacOS 14.7.2), and came across [this discussion](https://github.com/ghostty-org/ghostty/discussions/3723) that identified Quick Terminal as a possible culprit. Opening and closing the Quick Terminal reduced the energy usage back to expected levels for me.

**Reply by w3dd1e** [reply]
I saw that, but my battery drain was also happening with Ghostty app fully closed, not operating in the background. I uninstalled it and my battery went back to normal.

**Reply by mitchellh** [reply]
This is hard to believe, we don’t run any background code. I think something else caused this.

**Reply by w3dd1e** [reply]
I understand. If it would help, I'm happy to reinstall and try to replicate. Just let me know what info you would need, if so.

**Comment by anatskiy** [comment]
I can confirm the battery drain was indeed because of the Quick Terminal.

**Comment by micampe** [comment]
I don’t use the quick terminal, never enabled it, and I see the same high power drain.

**Reply by mitchellh** [reply]
Please provide concrete data we can use to chase this down if possible. An instruments capture like the one used to find the quick terminal issue would be fantastic.

**Comment by gmile** [comment]
Recently, my M3 Max battery discharge speed became noticeable, so I went to check what this might be & noticed a fairly large number next to Ghostty entry. See screenshot below: 

![CleanShot 2025-01-26 at 15 00 44@2x](https://github.com/user-attachments/assets/3a5d48dd-2735-4823-a204-1488b319f787)

"unfolding" the tree item doesn't reveal more detail with regards to which process might be eating the battery so extensively (there's more entries not shown on the screenshot, but their numbers are pretty much the same, e.g. near-zero battery usage):

![CleanShot 2025-01-26 at 15 08 06@2x](https://github.com/user-attachments/assets/6f1e002b-3a1f-473e-a92a-a85a35bfada1)

Any tips how to gather more details are appreciated! The version of Ghostty I am using is reported as "75fe0b2d".

**Reply by gmile** [reply]
**Upd**. The issue gone after updating from `75fe0b2d` to `1.0.1`:

![CleanShot 2025-01-27 at 01 10 57@2x](https://github.com/user-attachments/assets/6e33e86a-8f16-43f7-841a-3e9f074d7b6b)

**Comment by velicanu** [comment]
I don't think this issue is fixed. I am running the latest version, 1.1.0 

<img width="307" alt="Screenshot 2025-02-10 at 4 48 38 PM" src="https://github.com/user-attachments/assets/31225d58-066b-4e37-afe3-ef6825282ee2" />


and ghostty is still uses 10x more energy than the next biggest apps (web browsers).
<img width="1118" alt="Screenshot 2025-02-10 at 4 43 56 PM" src="https://github.com/user-attachments/assets/215a3d76-1a5f-4f51-b673-c574484dd300" />


I'm on an M4 mac. Happy to provide more info / logs / dig into anything.

**Reply by velicanu** [reply]
Noting that I've been trying to reproduce this behavior since I first encountered it and have been unable to do so. I don't know what caused Ghostty to have such high battery drain when it happened, but I have not been able to find a way to make it do so again. If I ever encounter this behavior again or figure out how to reliably reproduce it I'll update this thread.

**Comment by kfreezen** [comment]
Hey, I'd like to add that this is happening to me on both of my MacBooks. I am on 1.1.2 on my current MacBook. It seems to only have high energy impact when I'm using it. Whenever I check the stats on activity monitor, it's always zero, unless I've just been using it.

Working on downloading Xcode to run an instruments profile on my laptop. Will update you later on what I find.

**Reply by kfreezen** [reply]
https://www.dropbox.com/scl/fi/8d3ahg4oz6vect04i1hpr/ghostty-trace.trace.zip?rlkey=hb7cfx6dr7x3dgpvvih2vof1q&st=9qtpq82i&dl=0

Here is the link to a 1 minute and 30ish second trace with some foreground and background usage.

Hopefully this helps.

**Reply by mitchellh** [reply]
Thank you, just want to note I've downloaded this but haven't yet taken a look.

**Reply by kfreezen** [reply]
I noticed most of the weight shows up in thread_start with no further info. Maybe I need a debug build?

**Reply by kfreezen** [reply]
I just noticed today that iTerm is exhibiting the same high battery drain symptom on my MacBook Air.

macOS Sequoia 15.1.1.
MacBook Air M2, 2022

**Comment by seatedro** [comment]
Also experiencing this and it's staggering.
![image](https://github.com/user-attachments/assets/5ede36b0-5261-44d3-a707-f4bb83f0d557)
![image](https://github.com/user-attachments/assets/d466b7d5-8304-4481-88a8-09bffb4c9e15)
![CleanShot 2025-03-04 at 11 40 36@2x](https://github.com/user-attachments/assets/05697e58-0501-46e3-a47a-940580b354ea)

**Reply by mitchellh** [reply]
Note this report will also include anything running _within_ Ghostty, so unless this was Ghostty completely idle at a shell prompt, this data is not very helpful. See above how some people figured out it's the same between Ghostty and others because of this. 

We still haven't really identified any major power draw issues with Ghostty itself that are reproducible or traced that would allow us to resolve anything. (All reports so far have pointed to programs running in Ghostty instead)

**Reply by seatedro** [reply]
Understood! I will run my usual stuff on another terminal and report back in a day. In the meantime, if I could provide any traces/logs I would be happy to!

**Comment by cindy-postsig** [comment]
I had two servers running in two different Ghostty tabs -- maybe that's why it used so much energy.

<img width="682" alt="image" src="https://github.com/user-attachments/assets/7f411e8a-16d9-43e7-9b45-3acec944df81" />

**Comment by bbrodriges** [comment]
Hey, @mitchellh

I think I have found source of the problem. Please see two videos I've captured. Both have been capruted on Macbook Pro M3 Max, MacOS 15.3.1, Ghostty 1.1.2 (ee963f62).

First - https://drive.google.com/file/d/15HvtR8UZa9rL66GmclHwA929YFmT0eHG/view
Second - https://drive.google.com/file/d/12u3Pot6kD-_xCVUhRrsBkSfYDXtpSLy1/view

Both shows simple ghostty scroll back and forth, but in first video I have Pro Motion enabled on mac display, where in the second one I have set 60Hz refresh rate. As you can see in the first case power impact peaks at 150 in contrast with the second where power impact stays below 40, x4 power drain reduction.

**Comment by josevelaz** [comment]
![CleanShot 2025-06-09 at 16 56 39](https://github.com/user-attachments/assets/f789e400-1489-43c1-9985-e2a65b7320ba)
![CleanShot 2025-06-09 at 17 00 05](https://github.com/user-attachments/assets/973cdf3a-5299-46d0-bccd-afcd3deaa911)

Ive also been experiencing battery drainage for several weeks now. This used to not be an issue until recently

**Reply by mitchellh** [reply]
From above:

> Note this report will also include anything running within Ghostty, so unless this was Ghostty completely idle at a shell prompt, this data is not very helpful. See above how some people figured out it's the same between Ghostty and others because of this.

We'd need to identify actual resource usage (CPU, GPU) and how to reproduce those circumstances without running programs. People have successfully reported scenarios like this before, but note that an Activity Monitor report by itself is not helpful due to the above.

**Reply by josevelaz** [reply]
@mitchellh How can I help debug/trace this issue?

> but note that an Activity Monitor report by itself is not helpful due to the above.

I was not aware of this but its good to know!

**Comment by alxndr13** [comment]
chirping in here: 

- also high battery drain, most drain of all apps. 
- when nothing is rendered, the drain is basically zero. 
- But once the terminal renders something, the battery drain goes through the roof

**Comment by jondruse** [comment]
Same here.  I freakin love this terminal, but 

<img width="488" height="556" alt="Screenshot 2025-08-01 at 9 37 39 PM" src="https://github.com/user-attachments/assets/7a8eedda-ce54-466d-aa23-adc4968b8fa6" />
<img width="319" height="197" alt="Screenshot 2025-08-01 at 9 39 09 PM" src="https://github.com/user-attachments/assets/442b8065-afe7-4367-9677-8eab6d2fd0d1" />


I'm on an M4 MacBook Air

**Reply by mitchellh** [reply]
Please read my other messages. These screenshots are meaningless. This tracks Ghostty AND programs running WITHIN Ghostty. It doesn’t give maintainers any insight whatsoever into if what or how a problem exists. 

I’m going to close and lock this discussion. If a more actionable point of power usage can be found please open a new, directed discussion.
